### PR TITLE
Different Key and Value Strategies

### PR DESCRIPTION
**Pull request by stepanyanhayk**
PerDocMeanPooledRetrievalEncoder: 

- Keys: doc_key_embeddings — Qwen3 1024-dim vectors. Used only for FAISS retrieval; they never enter the forward pass of the model.
- Values: doc_tokens — the raw Qwen3 token IDs of the document. PerDocMeanPooledRetrievalEncoder takes these, looks them up in an nn.Embedding(vocab_size=152000, embedding_dim=128), and mean-pools over the token dimension per document: doc_tokens

LinearProjectionRetrievalEncoder:
- Keys: same as above, doc_key_embeddings, Qwen3 1024-dim, used only for FAISS. 
- Values: also doc_key_embeddings — same Qwen3 1024-dim vectors — but now passed through a trainable nn.Linear(1024 → 128)

### CODE REVIEW

File: src/medrap/conf/retrieval_encoder/per_doc_mean_pooled.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+_target_: medrap.retrieval_encoder.PerDocMeanPooledRetrievalEncoder
+vocab_size: 1024
+embedding_dim: 4
```
Comment (zaczw) [inline src/medrap/conf/retrieval_encoder/per_doc_mean_pooled.yaml]: Is this config meant to be toy/debug-only? The run script overrides this to vocab_size=151669 and embedding_dim=1024, while these defaults are 1024 and 4. If the larger values are the intended setting, would it make sense to align the defaults or naming/commenting this config as a debug config?

**Review by zaczw — APPROVED** [review]
